CHANGELOG.md

```diff
@@ -9,6 +9,63 @@ This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a
+- The `WordPress.Arrays.ArrayKeySpacingRestrictions` sniff has two new errorcodes: `TooMuchSpaceBeforeKey` and `TooMuchSpaceAfterKey`. Both auto-fixable.
+    The sniff will now check that there is _exactly_ one space on the inside of the square brackets around the array key for non-string, non-numeric array keys. Previously, it only checked that there was whitespace, not how much whitespace.
+- `WordPress.Arrays.ArrayKeySpacingRestrictions`: the fixers have been made more efficient and less fixer-conflict prone.
+- `WordPress.NamingConventions.PrefixAllGlobals`: plugin/theme prefixes should be at least three characters long. A new `ShortPrefixPassed` error has been added for when the prefix passed does not comply with this rule.
+- `WordPress.WhiteSpace.CastStructureSpacing` now allows for no whitespace before a cast when the cast is preceded by the spread `...` operator. This pre-empts a fixer conflict for when the spacing around the spread operator will start to get checked.
+- The `WordPress.WP.DeprecatedClasses` sniff will now detect classes deprecated in WP 4.9 and WP 5.3.
+- The `WordPress.WP.DeprecatedFunctions` sniff will now detect functions deprecated in WP 5.3.
+- `WordPress.NamingConventions.ValidHookName` now has "cleaner" error messages and higher precision for the line on which an error is thrown.
+- `WordPress.Security.EscapeOutput`: if an error refers to array access via a variable, the array index key will now be included in the error message.
+- The processing of the `WordPress` ruleset by `PHP_CodeSniffer` will now be faster.
+- Various minor code tweaks and clean up.
+- Various minor documentation fixes.
+- Documentation: updated the repo URL in all relevant places.
+
+### Deprecated
+- The `WordPress.WP.TimezoneChange` sniff. Use the `WordPress.DateTime.RestrictedFunctions` instead.
+    The deprecated sniff will be removed in WPCS 3.0.0.
+
+### Fixed
+- All sniffs in the `WordPress.Arrays` category will no longer treat _short lists_ as if they were a short array.
+- The `WordPress.NamingConventions.ValidFunctionName` and the `WordPress.NamingConventions.PrefixAllGlobals` sniff will now ignore functions marked as `@deprecated`.
+- Both the `WordPress.NamingConventions.PrefixAllGlobals` sniff as well as the `WordPress.WP.GlobalVariablesOverride` sniff have been updated to recognize variables being declared via (long/short) `list()` constructs and handle them correctly.
+- Both the `WordPress.NamingConventions.PrefixAllGlobals` sniff as well as the `WordPress.WP.GlobalVariablesOverride` sniff will now take a limited list of WP global variables _which are intended to be overwritten by plugins/themes_ into account.
+    Initially this list contains the `$content_width` and the `$wp_cockneyreplace` variables.
+- `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as it it were part of the hook name.
```

Comment:
```suggestion
- `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as if it were part of the hook name.
```

Reply:
@GaryJones Second opinion please. AFAICS the original sentence is correct English.

Reply:
As far as I know both were/was are valid here and it would be a matter of personal preference. I read this aloud about 20 times and it felt equally acceptable saying either version.

Reply:
Third person singular (he, she, it), subjunctive mood (imaginary, hypothetical) of "to be", means it should be:

```
- `WordPress.NamingConventions.ValidHookName`: will no longer examine a string array access index key as if it were a part of the hook name.
```

Reply:
I did consider the subjunctive here but it's also indicative and thus `was` becomes valid. For me I have no preference either way, both suited me :)

Reply:
This should be fixed now.